kernel/src/hil/time.rs

```diff
@@ -93,8 +113,33 @@ pub trait Client {
 /// The `Timer` trait models a timer that can notify when a particular interval
 /// has elapsed.
 pub trait Timer: Time {
+    fn set_client(&self, client: &'static TimerClient);
+
     /// Sets a one-shot timer to fire in `interval` clock-tics.
     fn oneshot(&self, interval: u32);
     /// Sets repeating timer to fire every `interval` clock-tics.
     fn repeat(&self, interval: u32);
+
+    // Q(alevy): Implementing this might require an additional, unnecessary field if a repeating
+    // timer is distinguished by having a non-zero value in the reload register. What if the return
+    // value is `Option<u32>` and `None` means it's oneshot, `Some(interval)` means it's repeating.
+    // Side benefit, `is_oneshot` and `is_repeating` can have default implementations.
+    fn interval(&self) -> u32;
+
+    fn is_oneshot(&self) -> bool;
+    fn is_repeating(&self) -> bool;
+
+    // This should return an option. Again, `is_enabled` can have a default implementation
+    fn time_remaining(&self) -> u32; // Returns 0 if disabled
```

Comment:
I'm confused that it's a comment rather than code, but agree with it? This would be better as `fn time_remaining(&self) -> Option<u32>` ?

Reply:
Actually, I might be more inclined to have it be a `Result<u32, ReturnCode>`? Kick back an `EOFF` if disabled?

Reply:
I think the comment is Amit's. I think that Option<u32> makes the most sense. If the ReturnCode is deterministic and there's only one there's no reason to have a ReturnCode.